tools/crm_mon.h

```diff
@@ -55,6 +55,15 @@ typedef enum mon_output_format_e {
     mon_output_cgi
 } mon_output_format_t;
 
+typedef struct {
+    bool is_error;
+
+    union {
+        stonith_history_t *history;
+        const char *error_msg;
+    } data;
+} mon_stonith_history_t;
```

Comment:
This sort of thing has come up for me at least once before on pacemaker, though I don't recall where at the moment.  A lot of the programming I do for myself is in Haskell, and that language has the concept of an option type.  A value of that type can either be Nothing or it can be Some <real type>.  That lets you differentiate between, say, an empty list and an invalid list that also just happens to be empty.

Anyway, that's kind of what I am going for here.  I'd prefer a more generic approach because I think the concept will come up again, too.